src/NQueen.java

```diff
@@ -0,0 +1,126 @@
+import java.util.Scanner;
+public class NQueen 
+{
+	public static void main(String args[]) throws Exception
+	{
+		Scanner scanner =new Scanner(System.in);
+		NQueen obj=new NQueen();
+		System.out.println("Enter the dimension of matrix : ");
+		int dimensionOfMatrix=scanner.nextInt();
+		scanner.close();
+		int result[][]=obj.createMatrix(dimensionOfMatrix);
+		NQueen.printResult(result);
+	}
+	
+	/**
+	 * @param dimensionOfMatrix
+	 * @return solution of N queen in 2D array
+	 * @throws Exception custom exception for n<=3
+	 */
+	public int[][] createMatrix(int dimensionOfMatrix)throws Exception
+	{
+		if(dimensionOfMatrix<=3)
+		{
+			throw new Exception("No Solution exist for N smaller than 3");
+		}
+		int board[][]=new int[dimensionOfMatrix][dimensionOfMatrix];
+		
+		Boolean result = nQueen(board , 0, dimensionOfMatrix);
+		if(result==true)
```

Comment:
It doesn't sense to compare Bboolean with Boolean here. you could simply do it like `if(result)`

Reply:
done